tools/windows-node-installer/test/e2e/azure_test.go

```diff
@@ -0,0 +1,226 @@
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"reflect"
+	"regexp"
+	"strings"
+	"testing"
+
+	"github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2019-03-01/compute"
+	"github.com/Azure/azure-sdk-for-go/services/network/mgmt/2019-04-01/network"
+	"github.com/Azure/go-autorest/autorest/azure"
+	"github.com/Azure/go-autorest/autorest/azure/auth"
+	"github.com/openshift/windows-machine-config-operator/tools/windows-node-installer/pkg/client"
+	"github.com/openshift/windows-machine-config-operator/tools/windows-node-installer/pkg/resource"
+)
+
+const (
+	// winRm Https port for the Windows node.
+	winRM = "5986"
+	// winRm protocol type.
+	winRMProtocol = "TCP"
+	// winRmPriority value
+	winRMPriority = 300
+	// winRM action type
+	winRmAction = "Allow"
+)
+
+// azureProvider stores Azure clients and resourceGroupName to access the windows node.
+type azureProvider struct {
+	// resourceGroupName of the Windows node
+	resourceGroupName string
+	// network security group client to check if winRmHttps port is opened or not.
+	nsgClient network.SecurityGroupsClient
+	// vm client to check for ansible ping on the windows node.
+	vmClient compute.VirtualMachinesClient
+}
+
+var (
+	// get the azure Auth Credentials specified from the env variable.
+	azureCredentials = os.Getenv("AZURE_AUTH_LOCATION")
+	// variable to be used for running the tests
+	azureInfo = azureProvider{}
+	// instanceIDs that are obtained from the windows-node-installer.json
+	instanceIDs []string
+	// secGroupIDs that are obtained from the windows-node-installer.json
+	secGroupsIDs []string
+)
+
+// TestWinRMSetup runs two tests i.e
+// 1. checks if the winRMHttps port is opened or not.
+// 2. ansible ping check to confirm that windows node is correctly
+//    configured to execute the remote ansible commands.
+func TestWinRMSetup(t *testing.T) {
+	t.Run("check if WinRmHttps port is opened in the inbound security group rules list", testWinRmPort)
+	t.Run("check if ansible is able to ping on the WinRmHttps port", testAnsiblePing)
+}
+
+// isNil is a helper functions which checks if the object is a nil pointer or not.
+func isNil(v interface{}) bool {
+	return v == nil || (reflect.ValueOf(v).Kind() == reflect.Ptr &&
+		reflect.ValueOf(v).IsNil())
+}
+
+// setup initializes the azureProvider to be used for running the tests.
+func setup() (err error) {
+	var vmClientPtr *compute.VirtualMachinesClient
+	var nsgClientPtr *network.SecurityGroupsClient
```

Comment:
Why are you defining these variables here? Please define them closer to the usage scope.